File: day01/day1.py
with open("day1.in") as f:
    calories = [[int(x) for x in l.strip().split("\n")] for l in f.read().split("\n\n")]

summed_cals = list(map(sum, calories))

print(f"Part 1: {max(summed_cals)}")

max3 = sum(sorted(summed_cals, reverse=True)[0:3])

print(f"Part 2: {max3}")

File: day01/day1.jl
elves = split(read("day1.in", String), "\n\n")

calories_per_elf = [parse.(Int, split(elf, "\n")) for elf in elves]

sum_per_elf = sum.(calories_per_elf)
top3 = partialsort(sum_per_elf, 1:3, rev=true) |> sum 

@show maximum(sum_per_elf)
@show top3;
